content/sklearn/concepts/clustering/clustering.md

```diff
@@ -0,0 +1,105 @@
+---
+Title: 'Clustering'
+Description: 'A technique that groups similar unlabled data points.'
+Subjects:
+  - 'Data Science'
+  - 'Machine Learning'
+Tags:
+  - 'Clustering'
+  - 'Machine Learning'
+  - 'Scikit-learn'
+  - 'Unsupervised learning'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+
+**Clustering** is an unsupervised machine learning technique. It groups similar rows of unlabled data. There are various clustering algorithms (like k-means,DBSCAN etc.). Different clustering algorithms apply different type of clustering. We will only be discussing the algorithms and types of clustering that can be implemented using sklearn.
+
+Clustering is more used as an exploratory data analysis.Each technique varies in their understanding of what a cluster constitutes and how to find them. So, for a dataset each algorithm may come up with different clusters. Since there is no truth against to verify, it's important to check the result. Until the quality of output is satisfactory, iterating to find the clusters are important.
+
+Each clusters are formed based on the similarity to each other. When the dataset has more and complex features the similarity measure becomes complex. After clustering, each cluster is given a unique cluster ID.
+
+Each clustering algorithm comes in two forms- a class and a function. The class implements the fit method to learn the clusters on training data. The function returns an array of integer labels corresponding to different clusters. 
+
+The 12 algorithms that can be executed using sklearn clustering are k-means,Affinity Propagation, Mean shift, Spectral clustering, Ward hierarchical clustering, Agglomerative clustering, DBSCAN, HDBSCAN, OPTICS, Gaussian mixtures, BIRCH, Bisecting k-means.
```

Comment:
```suggestion
**Clustering** is an unsupervised machine learning technique that groups similar rows of unlabeled data. Various clustering algorithms (such as k-means, DBSCAN, etc.) apply different types of clustering. 

Clustering is primarily used for exploratory data analysis. Each technique varies in understanding what constitutes a cluster and how to find them so that different algorithms may yield different clusters for the same dataset. Since there is no ground truth to verify against, it is essential to evaluate the results. Iterating to find satisfactory clusters is vital until the output quality meets expectations.

Each cluster is formed based on the similarity of its members. The similarity measures become more complicated as the dataset contains more complex features. After clustering, each cluster is assigned a unique cluster ID.

Each clustering algorithm is available in two forms: a class and a function. The class implements the fit method to learn the clusters from the training data, while the function returns an array of integer labels corresponding to different clusters.

The 12 algorithms that can be executed using `sklearn` for clustering are k-means, Affinity Propagation, Mean Shift, Spectral Clustering, Ward Hierarchical Clustering, Agglomerative Clustering, DBSCAN, HDBSCAN, OPTICS, Gaussian Mixtures, BIRCH, and Bisecting k-means.
```

Reply:
reframed for better readability